src/UI/Implementation/Component/Signal.php

```diff
@@ -15,12 +16,50 @@ class Signal implements \ILIAS\UI\Component\Signal {
 	protected $id;
 
 	/**
+	 * @var array
+	 */
+	protected $options = array();
+
+	/**
 	 * @param string $id
 	 */
 	public function __construct($id) {
 		$this->id = $id;
 	}
 
+	/**
+	 * @inheritdoc
+	 */
+	public function getId() {
+		return $this->id;
+	}
+
+	/**
+	 * @return array
+	 */
+	public function getOptions() {
+		return $this->options;
+	}
+
+	/**
+	 * @param string $key
+	 * @param mixed $value
+	 */
+	protected function addOption($key, $value) {
+		$this->options[$key] = $value;
+	}
+
+	/**
+	 * @param string $key
+	 * @return mixed|null
+	 */
+	protected function getOption($key) {
+		return (isset($this->options[$key])) ? $this->options[$key] : null;
+	}
```

Comment:
The options of a signal are sent as JSON object along with the javascript event, when the signal is getting triggered. See https://github.com/studer-raimann/ILIAS/blob/feature/5-3/ui-service-popovers-3/src/UI/Implementation/Render/AbstractComponentRenderer.php#L148
Thus, we need all properties represented as simple key/value pair. I'm not sure if this answers your question though ;-)